### DISCUSSION TITLE
InstaClient and Heroku

### DISCUSSION THREAD

**Discussion by efalloon** [davidwickerhf/instaclient#18] [Q&A] [answered]
I have been unable to launch my Heroku app with instaclient and receive a valid response. I am currently using the following line to create my client `client = InstaClient(driver_path="./drivers/chromedriver", localhost_headless=True)` but this will not work when attempting to login to an account.

I have heard that you need to use the following lines of code...
`chrome_options = webdriver.ChromeOptions()
  chrome_options.binary_location = os.environ.get("GOOGLE_CHROME_BIN")
  chrome_options.add_argument("--headless")
  chrome_options.add_argument("--disable-dev-shm-usage")
  chrome_options.add_argument("--no-sandbox")
  driver = webdriver.Chrome(executable_path=os.environ.get("./drivers/chromedriver"), chrome_options=chrome_options)`
... but how do I change the driver for the client to the following. Do you have any idea how to fix this.

**Comment by davidwickerhf** [comment, marked as answer]
I'm incredibly sorry for the late reply...
I have already setup the instaclient for Heroku, so you can set it directly through the initialization of the InstaClient object

You also need to do a bit of setup on Heroku:
You need to add 2 variables to your Heroku app's secrets via Heroku settings:

`heroku config:set GOOGLE_CHROME_BIN=/app/.apt/usr/bin/google_chrome`

`heroku config:set CHROMEDRIVER_PATH=/app/.chromedriver/bin/chromedriver`

Finally, you also need to add 2 buildbacks to the Heroku app:

`heroku buildpacks:add --index 1 https://github.com/heroku-buildpack-chromedriver`

`heroku buildpacks:add --index 2 https://github.com/heroku-buildpack-chromedriver`
